waku/waku_api/rest/admin/types.nim

```diff
@@ -0,0 +1,71 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  chronicles,
+  json_serialization,
+  json_serialization/std/options,
+  json_serialization/lexer
+import
+  ../serdes
+
+#### Types
+
+type
+  WakuPeer* = object
+    multiaddr*: string
+    protocol*: string
+    connected*: bool
+
+type WakuPeers* = seq[WakuPeer]
+
+#### Serialization and deserialization
+
+proc writeValue*(writer: var JsonWriter[RestJson], value: WakuPeer)
+  {.raises: [IOError].} =
+  writer.beginRecord()
+  writer.writeField("multiaddr", value.multiaddr)
+  writer.writeField("protocol", value.protocol)
+  writer.writeField("connected", value.connected)
+  writer.endRecord()
+
+proc readValue*(reader: var JsonReader[RestJson], value: var WakuPeer)
```

Comment:
Where are the procs `readValue` and `writeValue` used?

Reply:
These are called by JsonReader and JsonWriter classes as "template instantiation".